src/commands/commerce/extension/map.ts

```diff
@@ -0,0 +1,127 @@
+    public static example = [`sfdx ${CMD} --registered-extension-name --store-id --store-name`];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.map.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.map.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.map.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<string> {
+        this.ux.log(`Accessing Store using username: ${this.org.getUsername()} \n..........`);
+
+        return this.processMapExtension(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public processMapExtension(extensionName: string, storeName: string, storeId: string, userName: string): string {
+        const storeid = this.getStoreId(storeName, storeId, userName);
+        const registeredExternalServiceId = this.getExtensionName(extensionName, userName);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'StoreIntegratedService',
+            `Integration=${registeredExternalServiceId} StoreId=${storeid} ServiceProviderType='Extension'`,
+            userName
+        );
+
+        if (results instanceof SfdxError) {
+            throw new SfdxError(msgs.getMessage('extension.map.error', [extensionName, results.message]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeid, registeredExternalServiceId);
+    }
+
+    private getExtensionName(extensionName: string, userName: string): string {
+        let registeredExternalServiceId: string;
+        try {
+            registeredExternalServiceId = forceDataSoql(
+                `SELECT Id FROM RegisteredExternalService WHERE DeveloperName='${extensionName}'`,
+                userName
+            ).result.records[0].Id;
+        } catch (error) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            const errorMsg = msgs.getMessage('extension.map.nonexistant', [error.message]);
+            throw new SfdxError(errorMsg);
+        }
+        return registeredExternalServiceId;
+    }
+
+    private getStoreId(storeName: string, storeId: string, userName: string): string {
+        if (storeId === undefined) {
+            try {
+                storeId = forceDataSoql(`SELECT Id FROM WebStore WHERE Name='${storeName}' LIMIT 1`, userName).result
```

Comment:
As the store name is not unique, we decided to throw a descriptive error message when there is more than one store with the given name.
The error message can include the query/command to get all the stores with the given name
**example:** (not verified)
`More than on store found with the given store name. To see all the stores with the given name, run  "sfdx force:data:soql:query -q "SELECT ID, NAME FROM WEBSTORE WHERE NAME = 'XYZ'" --json `